src/utils/rootkeypackage_utils/src/rootkeypackage_parse.c

```diff
@@ -0,0 +1,827 @@
+ * @param[out] outPackage The root key package object to write parsed disabledSigningKeys data.
+ *
+ * @return The result.
+ */
+ADUC_Result
+RootKeyPackage_ParseDisabledSigningKeys(JSON_Object* protectedPropertiesObj, ADUC_RootKeyPackage* outPackage)
+{
+    ADUC_Result result = { .ResultCode = ADUC_GeneralResult_Failure, .ExtendedResultCode = 0 };
+
+    JSON_Array* hashesArray = NULL;
+    VECTOR_HANDLE hashes = NULL;
+    size_t count = 0;
+
+    if (protectedPropertiesObj == NULL || outPackage == NULL)
+    {
+        result.ExtendedResultCode = ADUC_ERC_UTILITIES_ROOTKEYPKG_UTIL_ERROR_BAD_ARG;
+        return result;
+    }
+
+    hashesArray = json_object_get_array(protectedPropertiesObj, "disabledSigningKeys");
+    if (hashesArray == NULL)
+    {
+        result.ExtendedResultCode = ADUC_ERC_UTILITIES_ROOTKEYPKG_PARSE_MISSING_REQUIRED_PROPERTY_DISABLEDSIGNINGKEYS;
+        goto done;
+    }
+
+    count = json_array_get_count(hashesArray);
+    if (count == 0)
+    {
+        result.ExtendedResultCode = ADUC_ERC_UTILITIES_ROOTKEYPKG_PARSE_DISABLEDSIGNINGKEYS_EMPTY;
+        goto done;
+    }
+
+    hashes = VECTOR_create(sizeof(ADUC_RootKeyPackage_Hash));
+    if (hashes == NULL)
+    {
+        result.ExtendedResultCode = ADUC_ERC_NOMEM;
+        goto done;
+    }
+
+    for (size_t i = 0; i < count; ++i)
+    {
+        ADUC_RootKeyPackage_Hash hashElement = { .alg = ADUC_RootKeyShaAlgorithm_INVALID, .hash = NULL };
+        ADUC_RootKeyShaAlgorithm tmpAlg = ADUC_RootKeyShaAlgorithm_INVALID;
+        CONSTBUFFER_HANDLE hashBuf = NULL;
+
+        JSON_Object* hashJsonArrayElementObj = json_array_get_object(hashesArray, i);
+        if (hashJsonArrayElementObj == NULL)
+        {
+            result.ExtendedResultCode = ADUC_ERC_UTILITIES_ROOTKEYPKG_PARSE_GETOBJ_DISABLEDSIGNINGKEYS_ELEMENT;
+            goto done;
+        }
+
+        result = RootKeyPackage_ParseShaHashAlg(hashJsonArrayElementObj, &tmpAlg);
+        if (IsAducResultCodeFailure(result.ResultCode))
+        {
+            goto done;
+        }
+
+        result = RootKeyPackage_ParseShaHashValue(hashJsonArrayElementObj, "hash", &hashBuf);
+        if (IsAducResultCodeFailure(result.ResultCode))
+        {
+            goto done;
+        }
+
+        hashElement.alg = tmpAlg;
+        hashElement.hash = hashBuf;
+
+        if (VECTOR_push_back(hashes, &hashElement, 1) != 0)
+        {
+            ADUC_RootKeyPackage_Hash_Free(&hashElement);
+
+            result.ExtendedResultCode = ADUC_ERC_NOMEM;
+            goto done;
+        }
+    }
+
+    (outPackage->protectedProperties).disabledSigningKeys = hashes;
+    hashes = NULL;
+
+    result.ResultCode = ADUC_GeneralResult_Success;
+
+done:
+    if (hashes != NULL)
+    {
+        size_t cnt = VECTOR_size(hashes);
+        for (size_t i = 0; i < cnt; ++i)
+        {
+            ADUC_RootKeyPackage_Hash* node = (ADUC_RootKeyPackage_Hash*)VECTOR_element(hashes, i);
+            ADUC_RootKeyPackage_Hash_Free(node);
```

Comment:
this won't destroy the individual nodes AFAIK. I believe Destroy only calls the free on the storage. Not the pointers within the storage:[ IMPLEMENT_MOCKABLE_FUNCTION(, VECTOR_HANDLE, VECTOR_move, VECTOR_HANDLE, handle)](https://github.com/Azure/azure-c-shared-utility/blob/4f1b7cd6bf48833fe4b9e23e2ff07510753faee5/src/vector.c#L56)

Reply:
Wrong one: https://github.com/Azure/azure-c-shared-utility/blob/4f1b7cd6bf48833fe4b9e23e2ff07510753faee5/src/vector.c#L41

Reply:
Correct. It recursively frees any member pointers pointing to allocations.  The VECTOR_destroy will destroy the contiguous memory for all the vector elements.  I've renamed to _DeInit suffix to make it clearer